static/js/main.js

```diff
@@ -96,75 +117,50 @@ $("document").ready(function() {
-    }
+    // Leaflet says it's more efficient to remove all the markers and then inseart the new ones.
     markers.clearLayers();
-    markers.addLayers(markerList);
+
+    // Let's put everything back onto the map
+    // We should also set the found attributes to false on the master list
+    var cleanMarkers = [];
+    for (var i = 0; i < allMarkers.length; i++) {
+      allMarkers[i].found = false;
+      // We are already looping over all the markers
+      // It might be more efficient to update cleanMarkers in this loop
+      cleanMarkers.push(allMarkers[i]);
+    }
+    markers.addLayers(cleanMarkers);
   }
 
-  function lookupValue(value) {
-    var val = $('#addressInput').val().toUpperCase();
-    var refreshNeeded = false;
-    var newMarkers = [];
+  function setMarkers() {
+    // Original design calls for putting all the markers back if there's nothing found during the search
+    // This feature is removed due to search slugishness
 
-    if (val.length > 3 && val.substring(0,3) === 'R72') {
-      for (var i = 0; i < allMarkerLength; i++) {
-        if (allMarkers[i].parcelid && allMarkers[i].parcelid.indexOf(val) >= 0) {
-          newMarkers.push(allMarkers[i]);
-        }
-      }
-      refreshNeeded = true;
-      markerList = newMarkers;
-    }
-    else if (val.length > 2 && val !== 'R' && val !== 'R7') {
-      for (var j = 0; j < allMarkerLength; j++) {
-        if (allMarkers[j].address && allMarkers[j].address.indexOf(val) >= 0) {
-          newMarkers.push(allMarkers[j]);
-        }
-      }
-      refreshNeeded = true;
-      markerList = newMarkers;
-    }
-    else if (val.length < 3 && markerList.length !== allMarkerLength) {
-      markerList = [];
+    // Let's find all the found markers 
+    var foundMarkers = [];
+    for (var i = 0; i < allMarkers.length; i++) {
+      if (allMarkers[i].found) foundMarkers.push(allMarkers[i]);
     }
+    // Let's not update the map if we found all markers in the search and they are already on the map
+    if (foundMarkers.length === allMarkers.length && markers.getLayers().length === allMarkers.length) return;
 
-    if (newMarkers.length === 0) {
-      refreshNeeded = true;
-      markerList = allMarkers;
-    }
-    else if (newMarkers.length === 1) {
-      selectedProperty(newMarkers[0].address, newMarkers[0].parcelid, newMarkers[0].claimed);
-    }
+    // Marker attributes such as .claim and .new can be used here to filter search results.
 
-    if (refreshNeeded) {
-      markers.clearLayers();
-      markers.addLayers(markerList);
+    // Leaflet says it's more efficient to remove all the markers and then inseart the new ones.
+    markers.clearLayers();
+    markers.addLayers(foundMarkers);
+
+    // Let's select the marker if there's only one of them is found
+    if (foundMarkers.length === 1 ) {
+      selectedProperty(foundMarkers[0].address, foundMarkers[0].parcelid, foundMarkers[0].claimed);
     }
   }
 
+  // NEEDS UPDATE: Broken by search update
```

Comment:
This should be relatively straight forward to re-implement. Places where you set the `found` flag could also take into account the state of the `new` flag.